src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -58,26 +58,18 @@ public static int Main(string[] args)
             AgentId = new Guid(args[0]);
             AgencyUrl = args[1];
             
-#if DEBUG
-            bool pause = false;
-#endif
             bool verbose = false;
             for (int i = 2; i < args.Length; i++)
                 switch (args[i])
                 {
-#if DEBUG
-                    case "--pause":
-                        pause = true;
+                    case "--debug-agent":
+                        if (!System.Diagnostics.Debugger.IsAttached)
+                            System.Diagnostics.Debugger.Launch();
```

Comment:
Have you considered adding a `Debugger.Break()` after the launch? Without it, execution will continue unless the developer has placed a break point